benchmarks/tabular_graph.cpp

```diff
@@ -16,6 +16,202 @@
  * https://arrow.apache.org/docs/cpp/csv.html
  */
 
+#include <cstring>     // `std::memchr`
+#include <algorithm>   // `std::search`
+#include <filesystem>  // Listing directories is too much pain in C
+#include <string_view> //
+#include <vector>      //
+#include <thread>      //
+#include <mutex>       //
+
+#include <benchmark/benchmark.h>
+#include <fmt/format.h>
+#include <argparse/argparse.hpp>
+
+#include <ukv/ukv.hpp>
+#include <dataset.h>
+
+namespace bm = benchmark;
+using namespace unum::ukv;
+
+constexpr size_t max_batch_size_k = 1024 * 1024 * 1024;
+constexpr ukv_str_view_t path_k = "./";
+
+static database_t db;
+
+static std::vector<size_t> source_sizes;
+static std::vector<std::string> source_files;
+std::mutex mtx;
```

Comment:
What kind if mutex is this? Is it static? What it does?

Reply:
I'm using ctime function which is not thread safe, in this case I'm use mutex.
I can place this mutex as static variable in function, if it better way of using it.

Reply:
My point is, having a global variable with name “mtx” is a code-smell. Anything is better.